src/tak/Game.java

```diff
@@ -208,6 +218,18 @@ void resign(Player p) {
         whenGameEnd();
     }
     
+    void setMarked(boolean marked, String field)
+    {
+        int file = field.charAt(0) - 'a';
+        int rank = field.charAt(1) - 1;
+        this.board[rank][file].setMarked(marked);
```

Comment:
Happens in the call, but can be included here, of course.
The regex only runs with [A-Ha-h] and [1-8].
But i will add it for security's sake.